odl/operator/operator.py

```diff
@@ -206,29 +395,64 @@ class Operator(with_metaclass(_OperatorMeta, object)):
     implementation of the respective other is provided.
     """
 
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance."""
+        instance = super().__new__(cls)
+
+        call_has_out, call_out_optional, _ = _dispatch_call_args(cls)
```

Comment:
I profiled the `examples/operator_test.py` (and fixed the diagnostics to work with the new syntax along the way), `__new__` takes 0.012 seconds for 162 calls (creations of the diagnostics functions as `FunctionSpace` elements). I guess that's acceptable?